spec/unit/graph/simple_graph_spec.rb

```diff
@@ -615,24 +619,25 @@ def graph_to_yaml(graph, which_format)
             expect(Set.new(vertices.keys)).to eq(Set.new(graph.vertices))
             vertices.each do |key, value|
               expect(value.keys.sort).to eq(%w{adjacencies vertex})
-              expect(value['vertex']).to equal(key)
+              expect(value['vertex']).to eq(key)
               adjacencies = value['adjacencies']
               expect(adjacencies).to be_a(Hash)
-              expect(Set.new(adjacencies.keys)).to eq(Set.new([:in, :out]))
+              expect(Set.new(adjacencies.keys)).to eq(Set.new(['in', 'out']))
               [:in, :out].each do |direction|
-                expect(adjacencies[direction]).to be_a(Hash)
+                direction_hash = adjacencies[direction.to_s]
+                expect(direction_hash).to be_a(Hash)
                 expected_adjacent_vertices = Set.new(graph.adjacent(key, :direction => direction, :type => :vertices))
-                expect(Set.new(adjacencies[direction].keys)).to eq(expected_adjacent_vertices)
-                adjacencies[direction].each do |adj_key, adj_value|
+                expect(Set.new(direction_hash.keys)).to eq(expected_adjacent_vertices)
+                direction_hash.each do |adj_key, adj_value|
                   # Since we already checked edges, just check consistency
                   # with edges.
                   desired_source = direction == :in ? adj_key : key
                   desired_target = direction == :in ? key : adj_key
                   expected_edges = edges.select do |edge|
                     edge['source'] == desired_source && edge['target'] == desired_target
                   end
-                  expect(adj_value).to be_a(Set)
-                  if object_ids(adj_value) != object_ids(expected_edges)
+                  expect(adj_value).to be_a(Array)
```

Comment:
Because `Set` is not `Data` and cannot be represented in YAML/JSON without special tags. The `Array` is unique though.